openquake/calculators/base.py

```diff
@@ -683,14 +691,20 @@ def build_riskinputs(self, kind, eps=numpy.zeros(0), eids=None):
                         if eps is not None and len(eps):
                             reduced_eps[ass.ordinal] = eps[ass.ordinal]
                 # build the riskinputs
+                if self.datastore.parent == ():
```

Comment:
Perhaps this should happen after checking `can_read_parent`? Maybe `can_read_parent` could be renamed into something like `get_parent` and return the parent, if available, or None otherwise?